include/libtorrent/torrent_info.hpp

```diff
@@ -580,7 +573,7 @@ namespace libtorrent
 		// this is a copy of the info section from the torrent.
 		// it use maintained in this flat format in order to
 		// make it available through the metadata extension
-		boost::shared_array<char> m_info_section;
+		std::unique_ptr<char[]> m_info_section;
```

Comment:
I didn't ask correctly, I mean now that `ut_metadata` is using `span` (not the interface), but I can infer the response: "keep the assert"